src/App.tsx

```diff
@@ -22,73 +22,128 @@ enum SortType {
 }
 
 type ReorderOptions = {
-  sortType: SortType,
-  isReversed: boolean,
+  sortType: SortType;
+  isReversed: boolean;
 };
 
-// Use this function in the render method to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
+export function getReorderedGoods(goods: string[],
+  { sortType, isReversed }: ReorderOptions) {
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort();
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
 export const App: React.FC = () => {
+  const [reorderOptions, setReorderOptions] = useState<ReorderOptions>({
+    sortType: SortType.NONE,
+    isReversed: false,
+  });
+  const [isButtonClicked, setIsButtonClicked] = useState(false);
+  const visibleGoods = getReorderedGoods(goodsFromServer, reorderOptions);
+
+  const handleSortAlphabetically = () => {
+    if (reorderOptions.sortType === SortType.ALPHABET) {
+      return;
+    }
+
+    setReorderOptions({
+      sortType: SortType.ALPHABET,
+      isReversed: false,
+    });
+    setIsButtonClicked(true);
+  };
+
+  const handleSortByLength = () => {
+    if (reorderOptions.sortType === SortType.LENGTH) {
+      return;
+    }
+
+    setReorderOptions({
+      sortType: SortType.LENGTH,
+      isReversed: false,
+    });
+    setIsButtonClicked(true);
+  };
+
+  const handleReverseOrder = () => {
+    setReorderOptions({
+      sortType: reorderOptions.sortType,
+      isReversed: !reorderOptions.isReversed,
+    });
+    setIsButtonClicked(true);
+  };
+
+  const handleResetOrder = () => {
+    setReorderOptions({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+    setIsButtonClicked(false);
+  };
+
+  const shouldShowResetButton = isButtonClicked;
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${reorderOptions.sortType === SortType.ALPHABET ? '' : 'is-light'}`}
```

Comment:
Use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally. Fix all cases